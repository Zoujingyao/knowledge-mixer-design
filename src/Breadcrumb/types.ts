import { ReactNode } from 'react';

export interface BreadcrumbProps {
  items: { key: any; label: ReactNode }[];
  separator?: string | ReactNode;
}
