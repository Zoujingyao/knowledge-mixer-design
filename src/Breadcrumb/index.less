.km-breadcrumb {
  a {
    color: rgba(0, 0, 0, 0.45);
    transition: color 0.3s;
  }
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  color: rgba(0, 0, 0, 0.85);
  font-size: 14px;
  font-variant: tabular-nums;
  line-height: 1.5715;
  list-style: none;
  font-feature-settings: 'tnum';
  > ol {
    display: flex;
    flex-wrap: wrap;
    margin: 0;
    padding: 0;
    list-style: none;
  }
  &-link {
    color: rgba(0, 0, 0, 0.45);
  }
  li:last-child .km-breadcrumb-link {
    color: rgb(0, 0, 0);
  }
  &-separator {
    margin: 0 8px;
    color: rgba(0, 0, 0, 0.45);
  }
  li:last-child .km-breadcrumb-separator {
    display: none;
  }
}
