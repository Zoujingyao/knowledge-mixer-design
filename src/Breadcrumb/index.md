---
title: Breadcrumb 面包屑
nav:
  title: Components
  path: /components
group:
  title: 布局
  path: /components/layout
---


# Breadcrumb 面包屑


## 基本用法

```tsx
import React from 'react';
import { Breadcrumb } from 'knowledge-mixer-design';

export default () => {
    const items = [
        { label: 'Home', key: 'item-1' }, // 菜单项务必填写 key
        { label: <a href="">Application Center</a>, key: 'item-2' },
        { label: <a href="">Application List</a>, key: 'item-3' },
        { label: 'An Application', key: 'item-4' },
    ];
  return (
    <>
      <Breadcrumb items={items}/>
    </>
  );
};
```


## 自定义分隔符

```tsx
import React from 'react';
import { Breadcrumb } from 'knowledge-mixer-design';

export default () => {
    const items = [
        { label: 'Home', key: 'item-1' }, // 菜单项务必填写 key
        { label: <a href="">Application Center</a>, key: 'item-2' },
        { label: <a href="">Application List</a>, key: 'item-3' },
        { label: 'An Application', key: 'item-4' },
    ];
  return (
    <>
      <Breadcrumb  separator='->' items={items}/>
    </>
  );
};
```