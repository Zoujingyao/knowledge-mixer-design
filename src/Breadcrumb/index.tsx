import React from 'react';
import { BreadcrumbProps } from './types';
import './index.less';
function Breadcrumb(props: BreadcrumbProps) {
  const { items, separator = '/' } = props;
  return (
    <nav className="km-breadcrumb">
      <ol>
        {items.map((el) => (
          <li key={el.key}>
            <span className="km-breadcrumb-link">{el.label}</span>
            <span className="km-breadcrumb-separator">{separator}</span>
          </li>
        ))}
      </ol>
    </nav>
  );
}

export default Breadcrumb;
