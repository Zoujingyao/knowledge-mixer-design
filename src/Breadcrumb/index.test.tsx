import React from 'react';
import Breadcrumb from './index';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

const items = [
  { label: 'Home', key: 'item-1' }, // 菜单项务必填写 key
  { label: <a href="">Application Center</a>, key: 'item-2' },
  { label: <a href="">Application List</a>, key: 'item-3' },
  { label: 'An Application', key: 'item-4' },
];

describe('<Breadcrumb />', () => {
  it('render Breadcrumb with dumi', () => {
    try {
      const ele = renderer.create(<Breadcrumb items={items} />).toJSON();
      expect(ele).toMatchSnapshot();
    } catch (e) {
      console.log(e);
    }
  });
});
