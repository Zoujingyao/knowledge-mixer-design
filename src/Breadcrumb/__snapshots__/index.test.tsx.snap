// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb /> render Breadcrumb with dumi 1`] = `
<nav
  className="km-breadcrumb"
>
  <ol>
    <li>
      <span
        className="km-breadcrumb-link"
      >
        Home
      </span>
      <span
        className="km-breadcrumb-separator"
      >
        /
      </span>
    </li>
    <li>
      <span
        className="km-breadcrumb-link"
      >
        <a
          href=""
        >
          Application Center
        </a>
      </span>
      <span
        className="km-breadcrumb-separator"
      >
        /
      </span>
    </li>
    <li>
      <span
        className="km-breadcrumb-link"
      >
        <a
          href=""
        >
          Application List
        </a>
      </span>
      <span
        className="km-breadcrumb-separator"
      >
        /
      </span>
    </li>
    <li>
      <span
        className="km-breadcrumb-link"
      >
        An Application
      </span>
      <span
        className="km-breadcrumb-separator"
      >
        /
      </span>
    </li>
  </ol>
</nav>
`;
